client/client_pool/include/client/client_pool/client_pool_timer.hpp

```diff
@@ -15,49 +15,45 @@
 
 #include <chrono>
 #include <future>
-#include <stdexcept>
 
-#include <boost/asio/deadline_timer.hpp>
-#include <boost/asio/io_service.hpp>
-#include <boost/date_time/posix_time/posix_time_duration.hpp>
-#include <boost/system/error_code.hpp>
+#include <asio/steady_timer.hpp>
+#include <asio/io_context.hpp>
 
 namespace concord_client_pool {
 template <typename ClientT>
 class Timer {
  public:
   using Clock = std::chrono::high_resolution_clock;
   Timer(std::chrono::milliseconds timeout, std::function<void(ClientT&&)> on_timeout)
-      : timeout_{timeout}, on_timeout_{on_timeout}, timer_(io_service_) {
+      : timeout_{timeout}, on_timeout_{on_timeout}, timer_(io_context_) {
     if (timeout_.count() > 0) {
-      timer_thread_future_ = std::async(std::launch::async, &Timer::WorkerThread, this);
+      timer_thread_future_ = std::async(std::launch::async, &Timer::work, this);
     }
   }
 
-  ~Timer() { stop(); }
+  ~Timer() { stopTimerThread(); }
 
-  void stop() {
+  void stopTimerThread() {
     if (timeout_.count() == 0) {
       return;
     }
-    stop_ = true;
-    io_service_.stop();
+    io_context_.stop();
     timer_thread_future_.wait();
   }
 
-  void set(const ClientT& client) {
-    if (timeout_.count() == 0) {
+  void start(const ClientT& client) {
+    if (timeout_.count() == 0 || not timer_thread_future_.valid() || io_context_.stopped()) {
       return;
     }
     client_ = client;
-    start_timer_ = std::chrono::steady_clock::now();
-    boost::posix_time::milliseconds timeout(timeout_.count());
-    timer_.expires_from_now(timeout);
-    auto handler = [this](const boost::system::error_code& error) {
-      if (error != boost::asio::error::operation_aborted) {
+    auto handler = [this](const asio::error_code& error) {
+      if (error != asio::error::operation_aborted) {
         on_timeout_(std::move(client_));
       }
     };
+
+    start_timer_ = std::chrono::steady_clock::now();
```

Comment:
Why did you move timer_ new settings to appear after a timeout handler?

Reply:
No strong reasoning, I wanted to group the code logically. Meaning, all timer and time value statements are clustered and separate from the handler and the client.